packages/blocks/src/__internal__/rich-text/keyboard.ts

```diff
@@ -405,15 +452,145 @@ export function createKeyboardBindings(page: Page, model: BaseBlockModel) {
         // if (context.format['code'] === true) {
         //   return ALLOW_DEFAULT;
         // }
+
+        // we need to insert text before show menu, because the Text node will be
+        // expired if we insert text after show menu because of the re-render
+        this.vEditor.insertText(range, context.event.key);
+        this.vEditor.setVRange({
+          index: range.index + 1,
+          length: 0,
+        });
+
         requestAnimationFrame(() => {
           const curRange = getCurrentNativeRange();
           showSlashMenu({ model, range: curRange });
         });
-        return ALLOW_DEFAULT;
+        return PREVENT_DEFAULT;
       },
     },
     ...createBracketAutoCompleteBindings(model),
   };
 
   return keyboardBindings;
 }
+
+export function createKeyDownHandler(
+  vEditor: AffineVEditor,
+  bindings: KeyboardBindings
+): (evt: KeyboardEvent) => void {
+  const bindingStore: Record<string, KeyboardBinding[]> = {};
+
+  const SHORTKEY = /Mac/i.test(navigator.platform) ? 'metaKey' : 'ctrlKey';
```

Comment:
Do we need to consider the iPad users?

```ts
export const safari = /Apple Computer/.test(navigator.vendor)
// `true` for both iOS and iPadOS
const ios = safari && (/Mobile\/\w+/.test(navigator.userAgent) || navigator.maxTouchPoints > 2)
```

Reply:
Are you referring to the scenario of using a keyboard on an iPad?

Reply:
> Are you referring to the scenario of using a keyboard on an iPad?

Yep